robot/impl/resource_manager.go

```diff
@@ -808,8 +808,11 @@ func (manager *resourceManager) completeConfigForRemotes(ctx context.Context, lr
 			}
 			rr, err := manager.processRemote(ctx, *remConf, gNode)
 			if err != nil {
-				gNode.LogAndSetLastError(
-					fmt.Errorf("error connecting to remote: %w", err), "remote", remConf.Name)
+				// If a remote is marked for removal then do not change status to unhealthy
+				if !gNode.MarkedForRemoval() {
```

Comment:
related to [the thread just below](https://github.com/viamrobotics/rdk/pull/4454/files#r1803669110) - an alternative approach is check if the node is unreachable before we mark it unhealthy.

```suggestion
				if !gNode.unreachable {
```

^ note that this code is for illustrative purposes and doesn't work as is - we would need to add a public getter for `.unreachable`.